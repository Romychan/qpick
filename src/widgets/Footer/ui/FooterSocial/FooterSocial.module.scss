.item {
  list-style-type: none;
}

.link {
  & svg {
    fill: var(--gray-1000);
    transition: 0.2s;
  }

  &:hover {
    color: var(--yellow-500);

    & svg {
      fill: var(--yellow-500);
    }
  }
}
