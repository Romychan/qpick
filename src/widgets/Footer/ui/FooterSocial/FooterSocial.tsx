import { Icon } from '~/shared/ui/Icon';
import { Menu } from '~/shared/ui/Menu';

import { FOOTER_SOCIAL_LINK } from '../../model/constants';

import styles from './FooterSocial.module.scss';

/** This component is used to display links to social media in `<Footer />` */
export const FooterSocial = () => {
  return (
    <Menu>
      {FOOTER_SOCIAL_LINK.map((item) => (
        <li key={item.link} className={styles.item}>
          <a
            href={item.link}
            aria-label={item.label}
            title={item.label}
            target="_blank"
            rel="noopener noreferrer"
            className={styles.link}
          >
            <Icon name={item.icon} size={30} />
          </a>
        </li>
      ))}
    </Menu>
  );
};
