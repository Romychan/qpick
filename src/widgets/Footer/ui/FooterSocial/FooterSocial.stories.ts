import type { Meta, StoryObj } from '@storybook/react';

import { FooterSocial } from './FooterSocial';

const meta = {
  title: 'Widgets/Footer/FooterSocial',
  component: FooterSocial,
  tags: ['autodocs'],
} satisfies Meta<typeof FooterSocial>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};
