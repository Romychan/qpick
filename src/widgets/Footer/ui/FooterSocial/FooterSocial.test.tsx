import { RenderResult, render, screen } from '@testing-library/react';

import { FooterSocial } from './FooterSocial';

describe('FooterSocial', () => {
  let container: RenderResult;

  beforeEach(() => {
    container = render(<FooterSocial />);
  });

  it('should render correctly', () => {
    const socialsElement = screen.getAllByTestId('icon');

    expect(socialsElement).toHaveLength(3);
  });

  it('should match snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
