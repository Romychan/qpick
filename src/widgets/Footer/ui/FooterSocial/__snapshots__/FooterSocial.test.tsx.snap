// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FooterSocial > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="_container_ca994d _horizontal_ca994d"
      >
        <ul
          class="_list_ca994d"
        >
          <li
            class="_item_f235a0"
          >
            <a
              aria-label="VK"
              class="_link_f235a0"
              href="https://vk.com/"
              rel="noopener noreferrer"
              target="_blank"
              title="VK"
            >
              <svg
                class="icon icon-vk-filled filled"
                data-testid="icon"
                height="30"
                width="30"
              >
                <use
                  xlink:href="/images/icons.svg#vk-filled"
                />
              </svg>
            </a>
          </li>
          <li
            class="_item_f235a0"
          >
            <a
              aria-label="Telegram"
              class="_link_f235a0"
              href="https://telegram.org/"
              rel="noopener noreferrer"
              target="_blank"
              title="Telegram"
            >
              <svg
                class="icon icon-telegram-filled filled"
                data-testid="icon"
                height="30"
                width="30"
              >
                <use
                  xlink:href="/images/icons.svg#telegram-filled"
                />
              </svg>
            </a>
          </li>
          <li
            class="_item_f235a0"
          >
            <a
              aria-label="Whatsapp"
              class="_link_f235a0"
              href="https://www.whatsapp.com/"
              rel="noopener noreferrer"
              target="_blank"
              title="Whatsapp"
            >
              <svg
                class="icon icon-whatsapp-filled filled"
                data-testid="icon"
                height="30"
                width="30"
              >
                <use
                  xlink:href="/images/icons.svg#whatsapp-filled"
                />
              </svg>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="_container_ca994d _horizontal_ca994d"
    >
      <ul
        class="_list_ca994d"
      >
        <li
          class="_item_f235a0"
        >
          <a
            aria-label="VK"
            class="_link_f235a0"
            href="https://vk.com/"
            rel="noopener noreferrer"
            target="_blank"
            title="VK"
          >
            <svg
              class="icon icon-vk-filled filled"
              data-testid="icon"
              height="30"
              width="30"
            >
              <use
                xlink:href="/images/icons.svg#vk-filled"
              />
            </svg>
          </a>
        </li>
        <li
          class="_item_f235a0"
        >
          <a
            aria-label="Telegram"
            class="_link_f235a0"
            href="https://telegram.org/"
            rel="noopener noreferrer"
            target="_blank"
            title="Telegram"
          >
            <svg
              class="icon icon-telegram-filled filled"
              data-testid="icon"
              height="30"
              width="30"
            >
              <use
                xlink:href="/images/icons.svg#telegram-filled"
              />
            </svg>
          </a>
        </li>
        <li
          class="_item_f235a0"
        >
          <a
            aria-label="Whatsapp"
            class="_link_f235a0"
            href="https://www.whatsapp.com/"
            rel="noopener noreferrer"
            target="_blank"
            title="Whatsapp"
          >
            <svg
              class="icon icon-whatsapp-filled filled"
              data-testid="icon"
              height="30"
              width="30"
            >
              <use
                xlink:href="/images/icons.svg#whatsapp-filled"
              />
            </svg>
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
